src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/split_explicit_free_surface.jl

```diff
@@ -0,0 +1,318 @@
+    timestepper = materialize_timestepper(free_surface.timestepper, maybe_extended_grid, free_surface, velocities, u_bc, v_bc)
+
+    return SplitExplicitFreeSurface(η,
+                                    barotropic_velocities,
+                                    filtered_state,
+                                    gravitational_acceleration,
+                                    kernel_parameters,
+                                    substepping,
+                                    timestepper)
+end
+
+# (p = 2, q = 4, r = 0.18927) minimize dispersion error from Shchepetkin and McWilliams (2005): https://doi.org/10.1016/j.ocemod.2004.08.002 
+@inline function averaging_shape_function(τ::FT; p = 2, q = 4, r = FT(0.18927)) where FT
+    τ₀ = (p + 2) * (p + q + 2) / (p + 1) / (p + q + 1)
+
+    return (τ / τ₀)^p * (1 - (τ / τ₀)^q) - r * (τ / τ₀)
+end
+
+@inline   cosine_averaging_kernel(τ::FT) where FT = τ ≥ 0.5 && τ ≤ 1.5 ? convert(FT, 1 + cos(2π * (τ - 1))) : zero(FT)
+@inline constant_averaging_kernel(τ::FT) where FT = convert(FT, 1)
+
+""" An internal type for the `SplitExplicitFreeSurface` that allows substepping with
+a fixed `Δt_barotropic` based on a CFL condition """
+struct FixedTimeStepSize{B, F}
+    Δt_barotropic    :: B
+    averaging_kernel :: F
+end
+
+""" An internal type for the `SplitExplicitFreeSurface` that allows substepping with
+a fixed number of substeps with time step size of `fractional_step_size * Δt_baroclinic` """
+struct FixedSubstepNumber{B, F}
+    fractional_step_size :: B
+    averaging_weights    :: F
+end
+
+function FixedTimeStepSize(grid;
+                           cfl = 0.7,
+                           averaging_kernel = averaging_shape_function,
+                           gravitational_acceleration = g_Earth)
+
+    FT = eltype(grid)
+
+    Δx⁻² = topology(grid)[1] == Flat ? 0 : 1 / minimum_xspacing(grid)^2
+    Δy⁻² = topology(grid)[2] == Flat ? 0 : 1 / minimum_yspacing(grid)^2
+    Δs   = sqrt(1 / (Δx⁻² + Δy⁻²))
+
+    wave_speed = sqrt(gravitational_acceleration * grid.Lz)
+
+    Δt_barotropic = convert(FT, cfl * Δs / wave_speed)
+
+    return FixedTimeStepSize(Δt_barotropic, averaging_kernel)
+end
+
+@inline function weights_from_substeps(FT, substeps, averaging_kernel)
+
+    τᶠ = range(FT(0), FT(2), length = substeps+1)
+    Δτ = τᶠ[2] - τᶠ[1]
+
+    averaging_weights = map(averaging_kernel, τᶠ[2:end])
+    idx = searchsortedlast(averaging_weights, 0, rev=true)
+    substeps = idx
+
+    averaging_weights = averaging_weights[1:idx]
+    averaging_weights ./= sum(averaging_weights)
+
+    return Δτ, tuple(averaging_weights...)
+end
+
+Base.summary(s::FixedTimeStepSize)  = string("Barotropic time step equal to $(prettytime(s.Δt_barotropic))")
+Base.summary(s::FixedSubstepNumber) = string("Barotropic fractional step equal to $(s.fractional_step_size) times the baroclinic step")
+
+Base.summary(sefs::SplitExplicitFreeSurface) = string("SplitExplicitFreeSurface with $(summary(sefs.substepping))")
```

Comment:
This summaries are not quite right. The default summary is something like `string(typeof(obj))`. A summary should always try to resemble the code, or how the object is constructed. These summaries are written word; there is no code. For example the summary for `FixedTimeStepSize` _must_ contain the string "FixedTimeStepSize".